Rectangle/Rectangle/RectangleStruct.cs

```diff
@@ -0,0 +1,88 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace RectangleStruct
+{
+    public struct Rectangle
+    {
+        private double width;
+        private double height;
+
+        public double Width
+        {
+            get => width;
+            set
+            {
+                Console.OutputEncoding = System.Text.Encoding.UTF8;
+                if (value <= 0)
+                    throw new ArgumentException("Ширина должна быть положительным числом");
+                width = value;
+            }
+        }
+
+        public double Height
+        {
+            get => height;
+            set
+            {
+                Console.OutputEncoding = System.Text.Encoding.UTF8;
+                if (value <= 0)
+                    throw new ArgumentException("Высота должна быть положительным числом");
+                height = value;
+            }
+        }
+
+        public double Area => Width * Height;
+        public double Perimeter => 2 * (Width + Height);
+
+        public Rectangle(double width, double height) : this()
+        {
+            Width = width;
+            Height = height;
+        }
+
+        public override string ToString() =>
+            $"Прямоугольник шириной {Width:F4} см и высотой {Height:F4} см";
+
+        public override bool Equals(object obj)
+        {
+            Console.OutputEncoding = System.Text.Encoding.UTF8;
+            if (!(obj is Rectangle))
```

Comment:
Тип данных не должен быть привязан к консоли
```suggestion
        {
            if (!(obj is Rectangle))
```